skipchain/skipchain_test.go

```diff
@@ -1104,29 +1109,25 @@ func TestRosterAddCausesSync(t *testing.T) {
 	log.Lvl1("Add last server into roster")
 	newBlock := &SkipBlock{
 		SkipBlockFix: &SkipBlockFix{
-			MaximumHeight: 2,
```

Comment:
Removing maxheight/baseheight here is just cleanup, right? Because they are set for "real" when creating the genesis block above?

When I was creating this test, I found that it was really fragile with respect to the config (max X base). We should wrap this in a loop that tries several different max/base configs.

Reply:
There are some tests that create blocks with different base/max-heights, but not with failing nodes. I will have to do that because of the comment I made above with forward-links where lower-level forward-links are missing.

Reply:
Added a test for a failing leader when creating secondary forward-links. Now we can have empty forward-links - check out with `ForwardLink.IsEmpty()`. If they were `nil`, protobuf removed them from the slice :(